src/telemetry/reportError.test.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2024 PixieBrix, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { getNotifier } from "webext-messenger";
+
+jest.mock("webext-messenger");
+
+describe("reportError", () => {
+  let consoleErrorSpy: jest.SpyInstance;
+  let _recordMock: jest.Mock;
+
+  let reportError: any;
+  beforeAll(() => {
+    _recordMock = jest.fn();
+    jest.mocked(getNotifier).mockReturnValue(_recordMock);
+
+    // Using require since the module uses an import-time initialization of the messenger which needs to be mocked
+    reportError = require("./reportError").default;
+  });
+
+  beforeEach(() => {
+    consoleErrorSpy = jest.spyOn(console, "error").mockImplementation();
+  });
+
+  afterEach(() => {
+    consoleErrorSpy.mockRestore();
+    jest.resetAllMocks();
+  });
+
+  it("should log error to console if logToConsole is true", () => {
+    const error = new Error("Test error");
+    reportError(error, { logToConsole: true });
+    expect(consoleErrorSpy).toHaveBeenCalledWith(error, { context: {} });
+  });
+
+  it("should not log error to console if logToConsole is false", () => {
+    const error = new Error("Test error");
+    reportError(error, { logToConsole: false });
+    expect(consoleErrorSpy).not.toHaveBeenCalled();
+  });
+
+  it("should not report error if it should be ignored", () => {
+    // "ResizeObserver loop limit exceeded" is an ignored error
+    const error = new Error("ResizeObserver loop limit exceeded");
```

Comment:
nit: (would remove the need for the comment)
```suggestion
    const shouldBeIgnoredError = new Error("ResizeObserver loop limit exceeded");
```